test/formulas/index.js

```diff
@@ -1,11 +1,57 @@
-require('../operators/operators');
-require('./math/math');
-require('./trigonometry/trigonometry');
-require('./text/text');
-require('./reference/reference');
-require('./information/information');
-require('./statistical/statistical');
-require('./date/date');
-require('./engineering/engineering');
-require('./logical/logical');
-require('./web/web');
+const FormulaError = require('../../formulas/error');
+const {FormulaParser} = require('../../grammar/hooks');
+const json = require('./TESTS.json')
+const expect = require('chai').expect;
+const sinon = require('sinon')
+
+const test = function(testName){
+    const date = new Date(2000, 1, 1)
+    let data = json[testName]["sheet"],
+        formula = json[testName]["formula"],
+        expected = json[testName]["expectedValue"]
+        clock =  sinon.useFakeTimers(date.getTime());
+    
+    if(data){
+        data.forEach((row, rowIndex) => row.forEach((value, colIndex) => {
+            if(value === "undefined")
+                data[rowIndex][colIndex] = undefined
+            if(FormulaError.errorMap.has(value))
+                data[rowIndex][colIndex] = FormulaError.errorMap.get(value)
+        }));
+    }
+    
+    const parser = new FormulaParser({
+        onCell: ref => {
+            return data[ref.row - 1][ref.col - 1]
+        },
+        onRange: ref => {
+            const arr = [];
+            for (let row = ref.from.row - 1; row < ref.to.row; row++) {
+                const innerArr = [];
+                for (let col = ref.from.col - 1; col < ref.to.col; col++) {
+                    innerArr.push(data[row][col])
+                }
+                arr.push(innerArr);
+            }
+            return arr;
+        },
+        onVariable: name => {
+            return json[testName]["variables"][name]
+        },
+    });
+
+    describe(testName, function () {
+        it(formula, () => {  
+            let result = parser.parse(formula, {row:1, col:1});
+            if (typeof result === "number" && typeof expected === "number") {
+                expect(result, `${formula} should equal ${expected}\n`).to.closeTo(expected, 0.00000001);
+            }else if (result.error)
+                // For FormulaError
+                 expect(expected === result.error, `${formula} should equal ${expected.error}\n`).to.equal(true);
+            else
+                expect(result, `${formula} should equal ${expected}\n`).to.equal(expected);
+        })
+    });
+}
+
+Object.keys(json).map((testName) => test(testName))
```

Comment:
You should guard this so it doesn't run by default. idk if you want this or something else as I'm unfamiliar with this part of jest.
https://stackoverflow.com/questions/4981891/node-js-equivalent-of-pythons-if-name-main